### PR TITLE
WalkingBadguy: intelligent badguys turn around at harmful tiles 

### LINKED ISSUE #851: Enemies that never fall off platforms should avoid spikes and harmful things on their path
@tobbi told me to make an issue on this. Enemies like bombs, smartballs, walking leaves etc. should avoid things like spikes and vines (turns around). This should be a property of the enemy so levels still work, but it by default they should turn around.

### PR DESCRIPTION
**Pull request by Nobelix**
This changes prevents intelligent walking bad guys from running into spikes etc...
Fixes #851

### CODE REVIEW

File: src/badguy/walking_badguy.hpp [hunk 1]
```diff
@@ -18,6 +18,7 @@
 #define HEADER_SUPERTUX_BADGUY_WALKING_BADGUY_HPP
 
 #include "badguy/badguy.hpp"
+#include "supertux/tile.hpp"
```
Comment (christ2go) [inline src/badguy/walking_badguy.hpp]: Do you need this include? Can it go into badguy.cpp?

File: src/badguy/walking_badguy.cpp [hunk 2]
```diff
@@ -190,6 +190,18 @@ WalkingBadguy::collision_badguy(BadGuy& , const CollisionHit& hit)
   return CONTINUE;
 }
 
+void
+WalkingBadguy::collision_tile(uint32_t tile_attributes)
+{
+  if(tile_attributes & Tile::HURTS && max_drop_height > -1 && on_ground())
```
Comment (christ2go) [inline src/badguy/walking_badguy.cpp]: Add space after if. Maybe add parantheses around first part for better readability.

File: src/badguy/walking_badguy.hpp (hunk 1 above)
Comment (Nobelix) [reply src/badguy/walking_badguy.hpp]: when it wasn't the compiler complained

File: src/badguy/walking_badguy.cpp (hunk 2 above)
Comment (Nobelix) [reply src/badguy/walking_badguy.cpp]: After what? I tried to mimic the other functions syntax

Comment (Nobelix) [reply src/badguy/walking_badguy.cpp]: Oh okay I get it

File: src/badguy/walking_badguy.hpp (hunk 1 above)
Comment (Nobelix) [reply src/badguy/walking_badguy.hpp]:
```/home/noe/Bureau/SuperTux/supertux/src/badguy/walking_badguy.cpp: In member function ‘virtual void WalkingBadguy::collision_tile(uint32_t)’:
/home/noe/Bureau/SuperTux/supertux/src/badguy/walking_badguy.cpp:196:26: error: ‘Tile’ has not been declared
   if( (tile_attributes & Tile::HURTS)  &&  max_drop_height > -1  &&  on_ground())

**Comment by tobbi** [comment]
This looks good, but needs a rebase. Can you please try to rebase this against current master so this can get merged?

**Pull request closed without merging**